examples/SimpleDrone/VirtualJoystick.py

```diff
@@ -0,0 +1,91 @@
+'''
+virtualjoystick.py
+This program will allow you to control solo's xy using an on screen joystick.
+It is designed to be run on a GCS.
+It utilizes SimpleDrone which allows for directional control using dronekit
+Note: It is very twitchy and will have kickback
+Programming exercise:
+Implement a better velocity decay
+'''
+import pygame
+from dronekit import *
+from SimpleDrone import SimpleDrone
+import time
+import math
+
+def draw_joystick(center,stick,radius):
+    #draw virtual joystick
+    screen.fill((0, 0, 0))
+    pygame.draw.line(screen, (0,0,255), center, stick, 2)
+    pygame.draw.circle(screen, (0,255,0), center, radius,2)
+    pygame.display.flip()
+
+
+SIM = True
+JOYSTICK_FEEL = 2.0
+DECAY = 0.7
+
+running = True
+width, height = 300,300
+radius = width/3
+down = False
+screen = pygame.display.set_mode((width,height))
+center = width/2,height/2
+joy = width/2,height/2
+last_decay = 0
+draw_joystick(center,joy,radius)
+
+print "connecting to drone..."
+if SIM:
+    vehicle = connect('127.0.0.1:14551', wait_ready=True)
+else:
+    vehicle = connect('0.0.0.0:14550', wait_ready=True)
+
+sd = SimpleDrone(vehicle, joystick_feel=JOYSTICK_FEEL)
+sd.take_control()
+
+if SIM:
```

Comment:
So are you assuming that on a real drone this will be flown to target area before executing?